propagation/honeycomb.go

```diff
@@ -0,0 +1,105 @@
+package propagation
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+// assumes a header of the form:
+
+// VERSION;PAYLOAD
+
+// VERSION=1
+// =========
+// PAYLOAD is a list of comma-separated params (k=v pairs), with no spaces.  recognized
+// keys + value types:
+//
+//  trace_id=${traceId}    - traceId is an opaque ascii string which shall not include ','
+//  parent_id=${spanId}    - spanId is an opaque ascii string which shall not include ','
+//  dataset=${datasetId}   - datasetId is the slug for the honeycomb dataset to which downstream spans should be sent; shall not include ','
+//  context=${contextBlob} - contextBlob is a base64 encoded json object.
+//
+// ex: X-Honeycomb-Trace: 1;trace_id=weofijwoeifj,parent_id=owefjoweifj,context=SGVsbG8gV29ybGQ=
+
+const (
+	TracePropagationHTTPHeader = "X-Honeycomb-Trace"
+	TracePropagationVersion    = 1
+)
+
+// MarshalHoneycombTraceContext uses the information in prop to create a trace context header
+// in the Honeycomb trace header format. It returns the header and header name as a key value
+// pair in a map.
+func MarshalHoneycombTraceContext(prop *PropagationContext) string {
```

Comment:
`prop` is a pointer to a `PropagationContext` so this code will panic when it gets to `prop.TraceContext` if it gets passed `nil`. 
(note that this was a pre-existing bug, so I'm glad that nobody's called this function with a nil PropagationContext before!)
```suggestion
func MarshalHoneycombTraceContext(prop *PropagationContext) string {
	if prop == nil {
		return ""
	}
```